pkg/controllers/elasticsearch/actions/create_nodepool.go

```diff
@@ -25,6 +26,35 @@ const (
 	esConfigVolumeMountPath = "/etc/pilot/elasticsearch/config"
 )
 
+type CreateNodePool struct {
+	Cluster  *v1alpha1.ElasticsearchCluster
+	NodePool *v1alpha1.ElasticsearchClusterNodePool
+}
+
+var _ controllers.Action = &CreateNodePool{}
+
+func (c *CreateNodePool) Name() string {
+	return "CreateNodePool"
+}
+
+func (c *CreateNodePool) Message() string {
+	return fmt.Sprintf("Created node pool %q", c.NodePool.Name)
+}
+
+func (c *CreateNodePool) Execute(state *controllers.State) error {
+	toCreate, err := nodePoolStatefulSet(c.Cluster, c.NodePool)
+	if err != nil {
+		return err
+	}
+
+	_, err = state.Clientset.AppsV1beta1().StatefulSets(toCreate.Namespace).Create(toCreate)
+	if err != nil {
+		return err
+	}
```

Comment:
Should we use  `IsAlreadyExistsError(err)` here and return nil in that case?
I feel like it might be good if `Execute` was guaranteed to be idempotent.
I feel like (not tested or verified) that there might be cases where we create the stateful set, but it doesn't get immediately listed and the reconciliation code elsewhere might calculate another create action.
In that case, we don't want to return an error because that will result in the cluster being put back in the queue.
Not sure, but worth discussing I think.

Reply:
I'd rather hold on this one for now. It's an overbearing requirement to impose on all actions, and (right now) there would no noticeable gain. Right now, if the call site wants to check if a `CreateNodePool` failed because it already exists, they can call IsAlreadyExists themselves and take appropriate action.

Once we start implementing more features with Actions, I hope it'll become clearer whether we want to require *all* actions be idempotent.